### PR TITLE
feat: allowedNetworks added to unity

### PR DESCRIPTION
**Pull request by cbartoszDell**
# Description
This PR adds `allowedNetworks`  parameter to CSM CR when Unity is being installed. This enables the user to select which networks should be used for node <-> array communication.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1222|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- Manually tested installation with the new fieid in `envs` section of CR, and without new field to make sure no regression happend.
- Ran unit tests.

### CODE REVIEW

File: operatorconfig/driverconfig/unity/v2.11.0/driver-config-params.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: <DriverDefaultReleaseName>-config-params
+  namespace: <DriverDefaultReleaseNamespace>
+data:
+  driver-config-params.yaml: |
+    CSI_LOG_LEVEL: "info"
```
Comment (abhi16394) [inline operatorconfig/driverconfig/unity/v2.11.0/driver-config-params.yaml:8]: @cbartoszDell could you add an extra parameter to this file after line 8
CSI_LOG_FORMAT: "JSON"
I recently made changes to n, n-1, n-2 version of unity.

File: samples/storage_csm_unity_v2110.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        # Specify kubelet config dir path.
+        # Ensure that the config.yaml file is present at this path.
+        # Default value: /var/lib/kubelet
+        - name: KUBELET_CONFIG_DIR
+          value: /var/lib/kubelet
+        # CSI_LOG_LEVEL is used to set the logging level of the driver.
+        # Allowed values: "error", "warn"/"warning", "info", "debug"
+        # Default value: "info"
+        - name: CSI_LOG_LEVEL
+          value: debug
```
Comment (abhi16394) [inline samples/storage_csm_unity_v2110.yaml:59]: @cbartoszDell could you add a line here as well to reflect the change that you would do  in cofigmaps?
# CSI driver log format
        # Allowed values: "TEXT" or "JSON"
        # Default value: "TEXT"
        - name: CSI_LOG_FORMAT
          value: "TEXT"
         or you can refer here for the change : https://github.com/dell/csm-operator/blob/d8c8867af12003c4058bf87a9e152719bed92ed7/samples/storage_csm_unity_v290.yaml#L63

File: operatorconfig/driverconfig/unity/v2.11.0/driver-config-params.yaml (hunk 1 above)
Comment (shefali-malhotra) [reply operatorconfig/driverconfig/unity/v2.11.0/driver-config-params.yaml:8]: addressed

File: samples/storage_csm_unity_v2110.yaml (hunk 2 above)
Comment (shefali-malhotra) [reply samples/storage_csm_unity_v2110.yaml:59]: addressed

**Review by shefali-malhotra — DISMISSED** [review]
LGTM

File: operatorconfig/driverconfig/unity/v2.11.0/upgrade-path.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+minUpgradePath: v2.9.1
```
Comment (rajendraindukuri) [inline operatorconfig/driverconfig/unity/v2.11.0/upgrade-path.yaml:1]: @shefali-malhotra Value here should be v2.10.0.. can you please check

Comment (shefali-malhotra) [reply operatorconfig/driverconfig/unity/v2.11.0/upgrade-path.yaml]: addressed

**Review by chimanjain — APPROVED** [review]

**Review by panigs7 — APPROVED** [review]

**Pull request merged**